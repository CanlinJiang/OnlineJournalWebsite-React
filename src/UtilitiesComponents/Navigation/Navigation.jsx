import React from 'react'
import {NavLink} from "react-router-dom";
import styles from "./Navigation.module.css";


export default function Navigation() {
    


    return (
        <nav className={styles.navigation}>
            <ul className={styles.leftBar}>
                <li ><NavLink to="/minelist" activeStyle={{textDecoration:"underline"}}>Mine's</NavLink></li>
                <li id ><NavLink to="/moodchoosing" activeStyle={{textDecoration:"underline"}}>Moods</NavLink></li>
                <li ><NavLink to="/randomwalk" activeStyle={{textDecoration:"underline"}}>Random Walk</NavLink></li>
            </ul>
            <ul className={styles.rightBar}> 
                <li id = "home" ><NavLink to="/">Home</NavLink></li>
                <li id = "addNewArticle" ><NavLink to="/addnewarticle">Add New Article</NavLink></li>
            </ul>
        </nav>
    )
}



