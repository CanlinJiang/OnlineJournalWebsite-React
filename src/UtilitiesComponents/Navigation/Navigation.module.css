.navigation {
  margin: 0;
  padding: 0;
  width: 100%;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  font-size: 1.05rem;
}

.leftBar {
    display: flex;
    flex-wrap: wrap;
    justify-content: flex-start;
}



.rightBar {
    display: flex;
    flex-wrap: wrap;
    justify-content: flex-end;
}



.navigation ul{
    list-style-type: none;
    margin: 0;
    padding: 0;
}

.navigation a{
    display: block;
    color: black;
    text-align: center;
    text-decoration: none;
    padding: 14px 16px;
    
}



.navigation a:hover{

    text-decoration: underline;
    
}

@media (max-width: 767px) {
    .navigation{
        display: grid;
        width: 100%;
        justify-content: center;
        align-items: center;
    }
    .leftBar{
        width: 100%;
        display: grid;
        grid-template-rows: 1fr 1fr 1fr;
        grid-template-columns: 1fr;
        justify-content: center;
        align-items: center;
        
    }
    .rightBar{
        order: -1;
        width: 100%;
        display: grid;
        grid-template-rows: 1fr 1fr;
        grid-template-columns: 1fr;
        justify-content: center;
        align-items: center;
    }
    .leftBar a {
        padding: 10px;

    }
    .rightBar a {
        padding: 10px;

    }
}