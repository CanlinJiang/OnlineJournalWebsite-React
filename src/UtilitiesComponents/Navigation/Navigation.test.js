import Navigation from "./Navigation"
import React from "react";

describe("Navigation tests", () => {
    it("renders correctly", () => {
       
        
      const { container } = render(<Navigation />);
      expect(container).toMatchSnapshot();
    });
  
  });