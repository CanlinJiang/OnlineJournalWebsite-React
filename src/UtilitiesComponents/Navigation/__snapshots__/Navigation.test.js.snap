// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation tests renders correctly 1`] = `
<div>
  <nav
    class="navigation"
  >
    <ul
      class="leftBar"
    >
      <li>
        <a
          href="/minelist"
        >
          Mine's
        </a>
      </li>
      <li>
        <a
          href="/moodchoosing"
        >
          Moods
        </a>
      </li>
      <li>
        <a
          href="/randomwalk"
        >
          Random Walk
        </a>
      </li>
    </ul>
    <ul
      class="rightBar"
    >
      <li
        id="home"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        id="addNewArticle"
      >
        <a
          href="/addnewarticle"
        >
          Add New Article
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
